src/app/communities/browse/page.tsx

```diff
@@ -0,0 +1,205 @@
+      }
+    }
+
+    void fetchTags()
+  }, []) // Runs only once when the component mounts
+
+  // Function to clear all filters and reset state
+  const clearFilters = () => {
+    setSelectedTags([])
+    setSearchQuery('')
+    setCurrentPage(1) // Reset to first page when filters are cleared
+  }
+
+  // useEffect to reset to the first page whenever filters or search query changes
+  useEffect(() => {
+    setCurrentPage(1)
+  }, [selectedTags, debouncedSearchQuery])
+
+  // Function to handle changes in search query
+  const handleSearchChange = (query: string) => setSearchQuery(query)
+
+  // Function to remove a tag from the selected tags list
+  const handleRemoveTag = (tagValue: string) => {
+    setSelectedTags(selectedTags.filter((tag) => tag.value !== tagValue))
+  }
+
+  // Function to clear the search query
+  const handleClearSearchQuery = () => setSearchQuery('')
+
+  return (
+    <div className="max-w-7xl p-6">
+      {/* Main title for the communities page */}
+      <h1 className="text-center font-subHeading text-h2 font-bold text-secondary-foreground">
+        Communities
+      </h1>
+
+      <div className="flex flex-col space-y-6 md:flex-row md:space-x-6 md:space-y-0">
+        {/* Search and tag filtering component */}
+        <SearchAndTagComponent
+          tags={tags}
+          selectedTags={selectedTags}
+          onSearchChange={handleSearchChange}
+          onTagChange={setSelectedTags}
+          onClearFilters={clearFilters}
+          searchQuery={searchQuery}
+        />
+        <div className="md:w-3/4">
+          {/* Display loading spinner while fetching communities */}
+          {isLoading && (
+            <div className="my-10 flex flex-col items-center justify-center">
+              <LoadingSpinner />
+              <p className="mt-4 text-muted">Loading communities...</p>
+            </div>
+          )}
+
+          {/* Display error message if there was an issue fetching communities */}
+          {hasError && (
+            <div className="my-10 text-center text-destructive">
+              <p>
+                Something went wrong while fetching the communities. Please try
+                again later.
+              </p>
+            </div>
+          )}
+
+          {/* Display communities if loading is complete and no error occurred */}
+          {!isLoading && !hasError && (
+            <>
+              {/* Show active filters if tags or search query are applied */}
+              {(selectedTags.length > 0 || searchQuery.trim()) && (
+                <ActiveFilters
+                  selectedTags={selectedTags}
+                  searchQuery={searchQuery}
+                  onRemoveTag={handleRemoveTag}
+                  onClearSearchQuery={handleClearSearchQuery}
+                />
+              )}
+
+              {/* Display community cards if there are communities available */}
+              {communities !== undefined &&
```

Comment:
```
communities !== undefined &&
communities !== null &&
```
These two conditions can be removed (communities is guaranteed to be an empty list or a list of valid communities)